modules/itc/src/main/scala/lucuma/itc/legacy/encoders.scala

```diff
@@ -0,0 +1,340 @@
+  import UnnormalizedSED._
+  a match {
+    case BlackBody(t)       =>
+      Json.obj(
+        "BlackBody" -> Json.obj(
+          "temperature" -> Json.fromDoubleOrNull(t.value.value.toDouble)
+        )
+      )
+    case PowerLaw(i)        =>
+      Json.obj("PowerLaw" -> Json.obj("index" -> Json.fromDoubleOrNull(i.toDouble)))
+    case StellarLibrary(s)  =>
+      Json.obj("Library" -> Json.obj("LibraryStar" -> Json.fromString(s.ocs2Tag)))
+    case s: CoolStarModel   =>
+      Json.obj("Library" -> Json.obj("LibraryStar" -> Json.fromString(s.ocs2Tag)))
+    case PlanetaryNebula(s) =>
+      Json.obj("Library" -> Json.obj("LibraryStar" -> Json.fromString(s.ocs2Tag)))
+    case Galaxy(s)          =>
+      Json.obj("Library" -> Json.obj("LibraryNonStar" -> Json.fromString(s.ocs2Tag)))
+    case Planet(s)          =>
+      Json.obj("Library" -> Json.obj("LibraryNonStar" -> Json.fromString(s.ocs2Tag)))
+    case HIIRegion(s)       =>
+      Json.obj("Library" -> Json.obj("LibraryNonStar" -> Json.fromString(s.ocs2Tag)))
+    case Quasar(s)          =>
+      Json.obj("Library" -> Json.obj("LibraryNonStar" -> Json.fromString(s.ocs2Tag)))
+    case _                  => // TODO UserDefined
+      Json.obj("Library" -> Json.Null)
+  }
+
+private given Encoder[Band] =
+  Encoder[String].contramap(_.shortName)
+
+private given Encoder[Redshift] =
+  Encoder.forProduct1("z")(_.z)
+
+given Encoder[ItcSourceDefinition] = (s: ItcSourceDefinition) =>
+  val source = s.profile match {
+    case _: SourceProfile.Point    =>
+      Json.obj("PointSource" -> Json.obj())
+    case _: SourceProfile.Uniform  => Json.obj("UniformSource" -> Json.obj())
+    case g: SourceProfile.Gaussian =>
+      Json.obj(
+        "GaussianSource" -> Json.obj(
+          "fwhm" -> Angle.signedDecimalArcseconds.get(g.fwhm).asJson
+        )
+      )
+  }
+
+  val units: Json = s.profile match {
+    case SourceProfile.Point(SpectralDefinition.BandNormalized(_, brightnesses))
+        if brightnesses.contains(s.normBand) =>
+      brightnesses.get(s.normBand).map(_.units.serialized) match {
+        case Some("VEGA_MAGNITUDE")                  => Json.obj("MagnitudeSystem" -> Json.fromString("Vega"))
+        case Some("AB_MAGNITUDE")                    => Json.obj("MagnitudeSystem" -> Json.fromString("AB"))
+        case Some("JANSKY")                          => Json.obj("MagnitudeSystem" -> Json.fromString("Jy"))
+        case Some("W_PER_M_SQUARED_PER_UM")          =>
+          Json.obj("MagnitudeSystem" -> Json.fromString("W/m²/µm"))
+        case Some("ERG_PER_S_PER_CM_SQUARED_PER_A")  =>
+          Json.obj("MagnitudeSystem" -> Json.fromString("erg/s/cm²/Å"))
+        case Some("ERG_PER_S_PER_CM_SQUARED_PER_HZ") =>
+          Json.obj("MagnitudeSystem" -> Json.fromString("erg/s/cm²/Hz"))
+        case _                                       =>
+          Json.Null
+      }
+    // FIXME Support emision lines
```

Comment:
I'm guessing this will be addressed in a future PR?

Reply:
Some day